### ISSUE TITLE
Update Library gives KeyError

### ISSUE THREAD

**Issue by davidvacek** [zewt/zDsonImport#2]
Hi Zewt,
first thank you a lot for this plugin.
I have an issue when zDsonImport>Update library

Error: KeyError: file C:\...\Documents\maya\scripts\zDsonImport-master\scripts\dsonimport\dson\content.py line 401: type #

When Importing I see nothing listed under Modifiers tab.
But Daz character imports... Also Apply HumanIk or Apply rigging works.

What is not imported are corrective blendshapes. Is it because Update library did not went trough the whole script?And because I see nothing under Modifiers tab when importing?

edit: I tried Maya 2017 and 2018

Thank you,
David

**Comment by zewt** [comment]
It looks like an attribute doesn't exist in whichever asset you're using that did in mine.  I updated the content code to allow for that.  It's not tested (don't have Daz installed right now), but give it a try.

**Comment by davidvacek** [comment]
Thank you for quick reply, looks the code runs bit further, but stops at
Error: KeyError: file C:\...\zDsonImport-master\scripts\dsonimport\dson\content.py line 315: asset_info 

I am getting lot of warnings in maya script window, I attached the output from script window

this one looks also bad :/
// Warning: Some weights could not be set to the specified value. The weight total would have exceeded 1.0. // 

# 
[mayaScriptOutput.txt](https://github.com/zewt/zDsonImport/files/2051882/mayaScriptOutput.txt)

**Comment by zewt** [comment]
Which base figure are you using?  I'd suggest starting with G3-based figures, since that's what this was written with, and once you have that working start trying other ones.

**Comment by davidvacek** [comment]
I am using G3 with some additional bought shaping features + Victoria 7
I just tried importing clean G3 without any additional stuff and gives me same / similar warnings

[mayaScriptOutput02.txt](https://github.com/zewt/zDsonImport/files/2051902/mayaScriptOutput02.txt)

**Comment by zewt** [comment]
The "KeyError... asset_info" looks like there's an asset that doesn't have the asset_info block.  I'm not sure why that would be missing, I added a check and a log so you can see which files are causing it.  If they're not assets you're importing it shouldn't matter.

The "weight total" warning is from skinning.  It might be innocuous (might just be rounding error for very slightly greater than 1 weights).  You can just move joints and check for any bad skin weighting to see if it's an issue.

The scale/general warnings happen because Daz3d has both scale X/Y/Z and a global scale, but Maya only has X/Y/Z, and I didn't want to complicate the resulting scenes to support both (most modifiers I tested with only used the component scale).  It should be pretty easy to tell if it's affecting the result.

**Comment by davidvacek** [comment]
Thank you for clearing it out for me. The main issues for me is that I am not getting corrective blendshapes. On attached image you can see what I am getting... Same I get if I import fbx to maya, with dae format it looks bit different, but still no corrective blendshapes for specific angles :/.

![weightnocorrectivemorphs](https://user-images.githubusercontent.com/34980776/40716352-fcce8292-6408-11e8-821f-0da7aef90e4b.jpg)

I can get from Daz blendshape meshes...but then need to setup them manually for every bone and angle that uses them with expression editor or set driven key :/. But as I understand plugin you created should reassign thees corrective blendshapes automatically?

**Comment by zewt** [comment]
In the import dialog, is the "DAZ 3D/Base Correctives" entry in the list and checked?  Make sure it's fully checked.  (It'll be grey if only some of the entries in the group are checked--by default it only checks modifiers that are required by other modifiers, but you want all of these.)  If you expand the group there should be a few dozen entries (if they're missing we should re-check the content library--are you still getting errors there?).

**Comment by davidvacek** [comment]
I am still getting Library Update error :( content.py line 315: asset_info # . I tried it today on different pc at work with maya 2017, getting same errors :(. And in the importer I see nothing listed :/
![screenshot_2](https://user-images.githubusercontent.com/34980776/40748181-df84a386-645f-11e8-96c3-6fe4c53e5959.jpg)

**Comment by zewt** [comment]
Did you update?  https://github.com/zewt/zDsonImport/commit/686c9a9791932cfc0067f5e12c4914ce7922e8e6

**Comment by davidvacek** [comment]
Yep at work on different pc I did clean plugin install and on my pc I updated content.py with latest but still getting the error :(

**Comment by zewt** [comment]
There's no asset_info on line 315, though.  Check the file and see if the change in https://github.com/zewt/zDsonImport/commit/686c9a9791932cfc0067f5e12c4914ce7922e8e6#diff-58ed97d6f56d9709c3e7ab714ca99da9 is there...

**Comment by davidvacek** [comment]
yep it is there, I've checked it, but actualy, it should create  log.info but I can't see any file like that

**Comment by davidvacek** [comment]
oh actually I am getting the log files now!

**Comment by davidvacek** [comment]
ok now when I try import no import dialog is shown

and this in log
[output.txt](https://github.com/zewt/zDsonImport/files/2055827/output.txt)

**Comment by zewt** [comment]
It looks like a circular dependency.  I'm not sure what modifier you're using that's causing that (doesn't happen here) so it's hard to test, but I added a check for it.

**Comment by davidvacek** [comment]
It works!!! if I select correctives manualy for example just for shin and thigh joints, it imports perfectly without any issues.

But if I check all the correctives in the import dialog:
![importdialog](https://user-images.githubusercontent.com/34980776/40753799-3c4c11ba-6476-11e8-8afc-0369d2252a39.jpg)

 I am getting (still imports the file but without connected corrective morphs)
[logImport.txt](https://github.com/zewt/zDsonImport/files/2056116/logImport.txt)

first I thought maybe there is some duplicate assets, but when I searched and checked it in import dialog, doesn't look like.

Thank you for helping me out! Is there possibility for donating? I Think you did perfect job with this plugin and also really helpful and quick support! I really appreciate it!
